.travis.yml

```diff
@@ -6,30 +6,81 @@
 # http://www.eclipse.org/legal/epl-v20.html
 #
 # SPDX-License-Identifier: EPL-2.0
-sudo: false
+sudo: true
 language: java
-cache:
-  apt: true
-  directories:
-    - $HOME/.m2
-before_cache:
-  - rm -rf $HOME/.m2/repository/org/eclipse/sw360/antenna/
-jdk:
-  - openjdk8
-  # - openjdk10
-  # - openjdk11
-  # - oraclejdk8
-  # - oraclejdk9
-  # - oraclejdk11
+addons:
+  apt:
+    packages:
+      - git
+# cache:
+#   apt: true
+#   directories:
+#     - $HOME/.m2
+# before_cache:
+#   - rm -rf $HOME/.m2/repository/org/eclipse/sw360/antenna/
+jdk: openjdk8
+
+before_install:
+  - sudo apt-get update -qq
+  - git clone --branch sw360-3.3.0-M1 --depth 1 https://github.com/eclipse/sw360 /tmp/sw360
+  - sudo /tmp/sw360/scripts/install-thrift.sh --no-cleanup
+  - pushd /tmp/sw360
+  - mvn install -pl backend/src/src-licenseinfo,libraries/lib-datahandler -am -DskipTests
+  - popd
+  - wget -N -O /tmp/install-jdk.sh https://raw.githubusercontent.com/sormuras/bach/ab648ed36892439f49a827cdf18c25e17c8095db/install-jdk.sh
+  - chmod +x /tmp/install-jdk.sh
+install:
+  - mvn install -DskipTests=true -B
 
 jobs:
   include:
-    - name: mvn test
-      install:
-        - mvn clean verify -P '!build-assembly' -DskipTests=true -V -B
-        - mvn install -P '!build-assembly' -DskipTests=true -B
+    # compile and execute with the same JDK
+    - &mvn-test
+      name: mvn test
+      script:
+        - mvn test -B
+        - cd example-projects/example-project
+        - mvn package
+    - <<: *mvn-test
+      name: mvn test openjdk9
+      jdk: openjdk9
+    - <<: *mvn-test
+      name: mvn test openjdk10
+      jdk: openjdk10
+    - <<: *mvn-test
+      name: mvn test openjdk11
+      jdk: openjdk11
+
+    # compile with JDK8 and execute with other JDK
+    - &differentJDK-test
+      name: mvn test, run with openjdk9
+      env:
+        - EXECUTION_JDK=9
       script:
-        - mvn test -P '!build-assembly' -B
+        - export JAVA_HOME=$HOME/custom_jdk
+        - /tmp/install-jdk.sh -f $EXECUTION_JDK -l GPL --target $JAVA_HOME
+        - cd example-projects/example-project
+        - mvn package -Dmaven.compiler.source=$EXECUTION_JDK -Dmaven.compiler.target=$EXECUTION_JDK
```

Comment:
Also use `clean` here.

Reply:
This is a fresh checked out subdirectory (`example-projects/example-project`) in which no mvn cmd was called before